File: web/src/model/victron_controller/dashboard/WeatherSocTemperatureSource.ts
// @ts-nocheck
import {BaboonCodecContext, BaboonBinWriter, BinTools, BaboonBinReader, Lazy} from '../../BaboonSharedRuntime'

export enum WeatherSocTemperatureSource {
    Forecast = "Forecast",
    Sensor = "Sensor"
}

export const WeatherSocTemperatureSource_values: ReadonlyArray<WeatherSocTemperatureSource> = [
    WeatherSocTemperatureSource.Forecast,
    WeatherSocTemperatureSource.Sensor
] as const;

export function WeatherSocTemperatureSource_parse(s: string): WeatherSocTemperatureSource {
    const found = WeatherSocTemperatureSource_values.find(v => v === s);
    if (found === undefined) {
        throw new Error("Unknown WeatherSocTemperatureSource variant: " + s);
    }
    return found;
}

export class WeatherSocTemperatureSource_UEBACodec {
    public encode(ctx: BaboonCodecContext, value: WeatherSocTemperatureSource, writer: BaboonBinWriter): unknown {
        if (this !== WeatherSocTemperatureSource_UEBACodec.lazyInstance.value) {
          return WeatherSocTemperatureSource_UEBACodec.lazyInstance.value.encode(ctx, value, writer)
        }
    
        switch (value) {
            case "Forecast": BinTools.writeByte(writer, 0); break;
                case "Sensor": BinTools.writeByte(writer, 1); break;
            default: throw new Error("Unknown enum variant: " + value);
        }
    }
    
    public decode(ctx: BaboonCodecContext, reader: BaboonBinReader): WeatherSocTemperatureSource {
        if (this !== WeatherSocTemperatureSource_UEBACodec .lazyInstance.value) {
            return WeatherSocTemperatureSource_UEBACodec.lazyInstance.value.decode(ctx, reader)
        }
    
        const tag = BinTools.readByte(reader);
        switch (tag) {
            case 0: return "Forecast" as WeatherSocTemperatureSource;
                case 1: return "Sensor" as WeatherSocTemperatureSource;
            default: throw new Error("Unknown enum variant tag: " + tag);
        }
    }

    public static readonly BaboonDomainVersion = '0.3.0'
    public baboonDomainVersion() {
        return WeatherSocTemperatureSource_UEBACodec.BaboonDomainVersion
    }
    public static readonly BaboonDomainIdentifier = 'victron_controller.dashboard'
    public baboonDomainIdentifier() {
        return WeatherSocTemperatureSource_UEBACodec.BaboonDomainIdentifier
    }
    public static readonly BaboonTypeIdentifier = 'victron_controller.dashboard/:#WeatherSocTemperatureSource'
    public baboonTypeIdentifier() {
        return WeatherSocTemperatureSource_UEBACodec.BaboonTypeIdentifier
    }

    protected static lazyInstance = new Lazy(() => new WeatherSocTemperatureSource_UEBACodec())
    public static get instance(): WeatherSocTemperatureSource_UEBACodec {
        return WeatherSocTemperatureSource_UEBACodec.lazyInstance.value
    }
}

File: crates/dashboard-model/src/victron_controller/dashboard/weather_soc_temperature_source.rs


#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, serde::Serialize, serde::Deserialize)]
pub enum WeatherSocTemperatureSource {
    Forecast,
    Sensor,
}

impl WeatherSocTemperatureSource {
    pub fn parse(s: &str) -> Result<Self, String> {
        match s {
            "Forecast" => Ok(WeatherSocTemperatureSource::Forecast),
            "Sensor" => Ok(WeatherSocTemperatureSource::Sensor),
            _ => Err(format!("Unknown variant: {}", s)),
        }
    }

    pub fn all() -> Vec<WeatherSocTemperatureSource> {
        vec![
            WeatherSocTemperatureSource::Forecast,
            WeatherSocTemperatureSource::Sensor,
        ]
    }
}

impl std::fmt::Display for WeatherSocTemperatureSource {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            WeatherSocTemperatureSource::Forecast => write!(f, "Forecast"),
            WeatherSocTemperatureSource::Sensor => write!(f, "Sensor"),
        }
    }
}

impl std::str::FromStr for WeatherSocTemperatureSource {
    type Err = String;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Self::parse(s)
    }
}

impl crate::baboon_runtime::BaboonBinCodecIndexed for WeatherSocTemperatureSource {
    fn index_elements_count(_ctx: &crate::baboon_runtime::BaboonCodecContext) -> u16 {
        0
    }
}

impl crate::baboon_runtime::BaboonBinEncode for WeatherSocTemperatureSource {
    fn encode_ueba(&self, _ctx: &crate::baboon_runtime::BaboonCodecContext, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        match self {
            WeatherSocTemperatureSource::Forecast => crate::baboon_runtime::bin_tools::write_byte(writer, 0)?,
            WeatherSocTemperatureSource::Sensor => crate::baboon_runtime::bin_tools::write_byte(writer, 1)?,
        }
        Ok(())
    }
}

impl crate::baboon_runtime::BaboonBinDecode for WeatherSocTemperatureSource {
    fn decode_ueba(_ctx: &crate::baboon_runtime::BaboonCodecContext, reader: &mut dyn std::io::Read) -> Result<Self, Box<dyn std::error::Error>> {
        let tag = crate::baboon_runtime::bin_tools::read_byte(reader)?;
        match tag {
            0 => Ok(WeatherSocTemperatureSource::Forecast),
            1 => Ok(WeatherSocTemperatureSource::Sensor),
            _ => Err(format!("Unknown enum variant tag: {}", tag).into()),
        }
    }
}